### PR TITLE
[2.3] [ClassLoader] removes deprecated classes from documentation.

### PR DESCRIPTION
**Pull request by hhamon**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

### CODE REVIEW

File: src/Symfony/Component/ClassLoader/ApcClassLoader.php [hunk 1]
```diff
@@ -17,13 +17,13 @@
...
- * other autoloader following this convention (the Composer one for instance)
+ * other autoloaders following this convention (the Composer one for instance).
  *
  *     $loader = new ClassLoader();
  *
  *     // register classes with namespaces
- *     $loader->add('Symfony\Component', __DIR__.'/component');
- *     $loader->add('Symfony',           __DIR__.'/framework');
+ *     $loader->addPrefix('Symfony\Component', __DIR__.'/component');
+ *     $loader->addPrefix('Symfony',           __DIR__.'/framework');
```
Comment (stof) [inline src/Symfony/Component/ClassLoader/ApcClassLoader.php]: This depends whether the ClassLoader above is expected to be the Symfony one or the Composer one (both are mentionned in the previous sentence)

**Comment by stof** [comment]
I would also add the missing ``@deprecated`` tags on the UniversalClassLoader (the tag is missing in the UniversalClassLoader class, even though the deprecation is mentionned in changelog files, and special flavors of the UniversalClassLoader were forgotten even in the changelog files even though we introduced the better alternatives a long time ago)

**Comment by hhamon** [comment]
There are no deprecations `2.3` for the `ClassLoader` component. They have been introduced in `2.4`. Should I put them anyway in the `2.3` branch?

**Comment by stof** [comment]
the replacement classes have been added in 2.1. We forgot to mark the old classes as deprecated at this time (otherwise they would have been removed in 2.3 already). Given that we tagged them only in 2.4, it might make sense to tag the forgotten ones in the 2.5 branch to be consistent.
But Composer was the recommended ClassLoader in 2.1+ projects anyway.

what do you think @fabpot ?

**Comment by fabpot** [comment]
I would tag the deprecated classes in 2.5.

**Comment by hhamon** [comment]
I have updated the documentation in this PR. Deprecation notices will be added to the 2.5 branch.

**Comment by hhamon** [comment]
Deprecation notices have been added to https://github.com/symfony/symfony/pull/13205

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @hhamon.

**Pull request merged**